smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java

```diff
@@ -0,0 +1,199 @@
+    private final Symbol outputSymbol;
+
+    private final String methodName;
+    private final String nonModularServiceName;
+    private final String paginationType;
+
+    PaginationGenerator(
+            Model model,
+            ServiceShape service,
+            OperationShape operation,
+            SymbolProvider symbolProvider,
+            TypeScriptWriter writer,
+            String nonModularServiceName
+    ) {
+
+        this.writer = writer;
+
+        this.serviceSymbol = symbolProvider.toSymbol(service);
+        this.operationSymbol = symbolProvider.toSymbol(operation);
+        this.inputSymbol = symbolProvider.toSymbol(operation).expectProperty("inputType", Symbol.class);
+        this.outputSymbol = symbolProvider.toSymbol(operation).expectProperty("outputType", Symbol.class);
+
+        String operationName = operation.getId().getName();
+        this.nonModularServiceName = nonModularServiceName;
+
+        // e.g. listObjects
+        this.methodName = Character.toLowerCase(operationName.charAt(0)) + operationName.substring(1);
+        this.paginationType = this.nonModularServiceName + "PaginationConfiguration";
+
+        PaginatedIndex paginatedIndex = model.getKnowledge(PaginatedIndex.class);
+        Optional<PaginationInfo> paginationInfo = paginatedIndex.getPaginationInfo(service, operation);
+        this.paginatedInfo = paginationInfo.orElseThrow(() -> {
+            return new CodegenException("Expected Paginator to have pagination information.");
+        });
+    }
+
+    @Override
+    public void run() {
+        // Import Service Types
+        writer.addImport(operationSymbol.getName(),
+                operationSymbol.getName(),
+                operationSymbol.getNamespace());
+        writer.addImport(inputSymbol.getName(),
+                inputSymbol.getName(),
+                inputSymbol.getNamespace());
+        writer.addImport(outputSymbol.getName(),
+                outputSymbol.getName(),
+                outputSymbol.getNamespace());
+        String nonModularLocation = serviceSymbol.getNamespace()
+                .replace(serviceSymbol.getName(), nonModularServiceName);
+        writer.addImport(nonModularServiceName,
+                nonModularServiceName,
+                nonModularLocation);
+        writer.addImport(serviceSymbol.getName(), serviceSymbol.getName(), serviceSymbol.getNamespace());
+
+        // Import Pagination types
+        writer.addImport("Paginator", "Paginator", "@aws-sdk/types");
+        writer.addImport(paginationType, paginationType, "./" + PAGINATION_INTERFACE_FILE.replace(".ts", ""));
+
+        this.writeClientSideRequest();
+        this.writeRequest();
+        this.writePager();
+    }
+
+    static String getOutputFilelocation(OperationShape operation) {
+        return "pagination/" + operation.getId().getName() + "Paginator.ts";
+    }
+
+    static void generateServicePaginationInterfaces(String nonModularServiceName,
```

Comment:
Can you follow [the guidance here](https://github.com/awslabs/smithy-typescript/pull/200#discussion_r474197159) for this signature as well?